chore: Bump Poison to 4.0

cmschuetz opened the pull request:
- Bumps Poison to 4.0
- Updates APIs to work with the new version
- Autoformatter changes

Inspired by: https://github.com/wfgilman/plaid-elixir/commit/20b137436a2386771c9519e8a9fc8120ea4f907f

cmschuetz committed:
chore: Bump Poison to 4.0

- Bumps Poison to 4.0
- Updates APIs to work with the new version
- Autoformatter changes

wfgilman commented on lib/dwolla.ex:
```diff
@@ -101,30 +103,31 @@ defmodule Dwolla do
-      {:ok, parsed_body} -> parsed_body
-      {:error, _}        -> {:invalid, body}
-    end
+    Poison.Parser.parse!(body, %{})
```
@cmschuetz why the choice to raise on failure to parse instead of returns a tuple?

cmschuetz committed:
fix: New Oauth Token flow

- Replaces old oauth token flow with requests to "#{root_uri}/token" as the existing route is no longer available
- Bumps a few dependencies

cmschuetz committed:
fix: Remove ip_address as a required param

- Removes ip_address as a required param when creating a user

cmschuetz committed:
feat: Add bank_account_type to FundingSource

- Adds bank_account_type to the FundingSource struct to ensure this gets deserialized from dwolla responses

cmschuetz committed:
Merge branch 'master' of github.com:wfgilman/dwolla-elixir

wfgilman reviewed (commented):
Thanks for the PR! I'm happy to merge but first wanted to understand the choice to raise with `parse!/2`? Sometimes Dwolla will return an HTML blob during 500 errors which causes the decoding to fail.

cmschuetz committed:
fix: parse! -> decode

cmschuetz commented on lib/dwolla.ex:
```diff
@@ -86,10 +86,11 @@ defmodule Dwolla do
+
   def process_response_body(body) do
-    case Poison.Parser.parse(body) do
+    case Poison.decode(body) do
```
@wfgilman Thanks for the callout, updated this to `decode` as this seems to replace the behavior of `Poison.Parser.parse` in v4
